takwimu/management/commands/update_topics_index.py

```diff
@@ -25,45 +34,127 @@ def index_topics(self, queryset, search_backend):
+                                self.stdout.write(
+                                    search_backend.es_index + ": Indexing widget '%s result %s'" % (
+                                        data['id'],
+                                        outcome,
+                                    ))
+
+    def index_hurumap(self, search_backend):
+        options = webdriver.ChromeOptions()
+        options.add_argument('headless')
+        # chrome can not be run as root, which is the case in docker
+        options.add_argument('no-sandbox')
+        options.add_argument('disable-gpu')
+        browser = webdriver.Chrome(options=options)
+
+        server_url = os.getenv('HURUMAP_URL', 'http://localhost:8000')
+
+        if server_url.endswith('/'):
+            # remove trailing slash
+            server_url = server_url[:-1]
+
+        urls = {'Kenya': 'profiles/country-KE-kenya/',
+                'Uganda': 'profiles/country-UG-uganda/',
+                'Ethiopia': 'profiles/country-ET-ethiopia/',
+                'Tanzania': 'profiles/country-TZ-tanzania/',
+                'Nigeria': 'profiles/country-NG-nigeria/',
+                'Senegal': 'profiles/country-SN-senegal/',
+                'Burkina Faso': 'profiles/country-BF-burkina-faso/',
+                'Democratic Republic of Congo': 'profiles/country-CD-democratic-republic-of-congo',
+                'Zambia': 'profiles/country-ZM-zambia/',
+                'South Africa': 'country-ZA-south-africa/'
+                }
+
+        for country, url in urls.items():
+            self.stdout.write(f"Working on {country} {server_url}/{url}")
+            browser.get(f"{server_url}/{url}")
+            soup = BeautifulSoup(browser.page_source, 'lxml')
+            visualizations = soup.find_all('div', {
+                "class": ["column-full", "column-half", "column-three-quarters",
+                          "column-quarter"]})
+            self.stdout.write(
+                f"Found {len(visualizations)} possible visualizations")
+            for viz in visualizations:
+                chart_header = ''
+                try:
+                    chart_header = viz.find('h3', class_='chart-header').text
+                except AttributeError:
+                    pass
+
+                title = viz.get('data-chart-title', '') or chart_header
+
+                # if section does not have a title then it's probably a random
+                # div without a visualization
+
+                if title and viz.find('div', class_="chart"):
+                    # self.stdout.write(str(viz))
+                    labels = ' '.join(
+                        [i.text for i in viz.select('span.x.axis.label')])
+
+                    qualifiers = ' '.join(
+                        [i.text for i in
+                         viz.find_all('span', class_=['chart-qualifier'])])
+
+                    id = f"{country}-{viz['id']}"
+                    link = f"{server_url}/{url}#{viz['id']}"
```

Comment:
Use relative urls for internal links... full urls could end up being opened in new tab in the front end (depending on component used).